src/Engine/Objects/Actor.cpp

```diff
@@ -3309,10 +3309,11 @@ void Actor::DamageMonsterFromParty(signed int a1, unsigned int uActorID_Monster,
 
             default:
                 attackElement = (DAMAGE_TYPE)player->GetSpellSchool(
-                    projectileSprite->spell_id);
+                    static_cast<SPELL_TYPE>(projectileSprite->spell_id));
```

Comment:
This is the code in `class SpriteObject`:

```cpp
    int16_t field_22_glow_radius_multiplier = 1;
    ItemGen containing_item;
    int spell_id = 0;
    int spell_level = 0;
    PLAYER_SKILL_MASTERY spell_skill = PLAYER_SKILL_MASTERY_NONE;
    int field_54 = 0;
```

Replace it with:
```cpp
    int16_t field_22_glow_radius_multiplier = 1;
    ItemGen containing_item;
    SPELL_TYPE spell_id = SPELL_NONE;
    char pad[3] = {0, 0, 0}; // TODO(yourhandle): add SpriteObject_MM7 to LegacyImages and redo this properly.
    int spell_level = 0;
    PLAYER_SKILL_MASTERY spell_skill = PLAYER_SKILL_MASTERY_NONE;
    int field_54 = 0;
```

Then get rid of casts.

The next step would be to resolve the TODO (in a separate PR). You can do this by:
1. Introducing `SpriteObject_MM7` class in `LegacyImages.h` that matches the binary layout used in MM7 engine.
2. Dropping the padding from `class SpriteObject`.
3. Adding the corresponding `Serialize` / `Deserialize` functions (look around the `LegacyImages.cpp` to see how it's done).
4. Using `SpriteObject_MM7` in loading / saving code. There are uses in `SaveLoad.cpp` and maybe in some other places.
5. Get rid of `memcpy` calls for `SpriteObject`, just copy the values of this struct directly.